src/platform/native/Core.zig

```diff
@@ -310,6 +319,29 @@ fn initCallbacks(self: *Core) void {
     }.callback;
     self.window.setScrollCallback(scroll_callback);
 
+    const joystick_callback = struct {
+        fn callback(joystick: glfw.Joystick, event: glfw.Joystick.Event) void {
+            const pf = core_instance.?;
```

Comment:
Why is a global variable being used here, instead of `Joystick.setUserPointer` and `Joystick.getUserPointer` being used to store a pointer `*Core`?

Reply:
Because from the GLFW joystick call back it's impossible to access `*Core` in order to set the joystick user pointer when it is connected. 

The other way to doing this is checking all 16 possible joysticks to see if one is connected (using `Joystick.present`) instead of using the callback. But `present` can perfom joystick initialization and doing these checks every frame might not be preferable.

Reply:
Ah I get it - nevermind - it's because this _is_ the connect/disconnect callback, and there's no way to pass context into here. This approach is fine.